### PR TITLE
Truckparser - final edition

### PR DESCRIPTION
**Pull request by only-a-ptr**
Although it's not visible in the commit log (most bugfix commits were squashed...)

# ... the hero of bug-fixing is Ulteq :clap: 

This branch is almost ready for merge; some very minor glitches remain.

### CODE REVIEW

**Comment by ulteq** [comment]
[Turboprops](http://docs.rigsofrods.org/vehicle-creation/fileformat-truck/#turboprops) are not parsed correctly.

Reference: https://github.com/only-a-ptr/ror-legacy-svn-trunk/blob/0a55a993899d6a06cdc58fd98b5b675b371eb89a/source/main/physics/input_output/SerializedRig.cpp#L3938-L3978

**Comment by ulteq** [comment]
The `Piper J3 Super Cab` segfaults in `FinalizeRig()`

**Comment by only-a-ptr** [comment]
Found a minor GFX glitch: when paused, the "spinning pistonprop" animation appear even if the airplane is off.
ror-glitch-pause-pistonprop

**Comment by ulteq** [comment]
What about this: https://github.com/RigsOfRods/rigs-of-rods/issues/686

**Comment by tritonas00** [comment]
https://github.com/RigsOfRods/rigs-of-rods/pull/1137/commits/1db174c5bb2535255e2786d9e2e06b89a27dcd2a fixes Piper J3 segfault but breaks wings with other airplanes:
2016-10-08-140524_1366x768_scrot

log: http://pastebin.com/2BQW2cn7

**Comment by ulteq** [comment]
Dropping this commit: https://github.com/RigsOfRods/rigs-of-rods/pull/1137/commits/b3e7a2e8743f02180fd1535e4cc08b3be71cde97 should resolve the merge conflict.

I replaced the use-counter with a simple boolean flag: https://github.com/RigsOfRods/rigs-of-rods/pull/1142/commits/cd010683065de984cef2f13824e12aba758a425a

File: source/main/physics/RigSpawner.cpp [hunk 1]
```diff
@@ -1063,112 +1062,84 @@ void RigSpawner::ProcessAirbrake(RigDef::Airbrake & def)
...
-		return;
-	}
+    // Perform checks
+    if (! this->CheckWingLimit(1)) { return; }
 
-	std::stringstream wing_name;
-	wing_name << "wing-" << m_rig->truckname << "-" << m_rig->free_wing;
-	std::stringstream wing_obj_name;
-	wing_obj_name << "wingobj-" << m_rig->truckname << "-" << m_rig->free_wing;
+    if ((m_first_wing_index != -1) && (m_rig->wings[m_rig->free_wing - 1].fa == nullptr))
```
Comment (ulteq) [inline source/main/physics/RigSpawner.cpp]: Are you sure that `m_rig->free_wing` is always greater than zero?

**Comment by ulteq** [comment]
This should fix the wing issue: https://github.com/ulteq/rigs-of-rods/commit/03879cf2098bdc1cb24968a78060c3f2402d6e79

File: source/main/physics/RigSpawner.cpp (hunk 1 above)
Comment (only-a-ptr) [inline source/main/physics/RigSpawner.cpp]: Initial state is `m_first_wing_index = -1; free_wing = 0;`
If `m_first_wing_index != -1`, there must be at least 1 wing.

Comment (ulteq) [inline source/main/physics/RigSpawner.cpp]: Ok, let's just leave it like that.

**Comment by only-a-ptr** [comment]
@tritonas00 I have absolutely no clue how could _that_ have hapenned. I only reordered tasks regarding wing processing, which might have caused wings to stop working. But this... :confused:

**Comment by tritonas00** [comment]
https://github.com/ulteq/rigs-of-rods/commit/03879cf2098bdc1cb24968a78060c3f2402d6e79 fixes the issue, i just tried it.

**Comment by only-a-ptr** [comment]
I cherry-picked all fixes from Ulteq's branch. Most got squashed into the original faulty commits.

Merging in 3... 2... 1...

**Pull request merged**